src/views/Teams/Team.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react'
+import Page from 'components/layout/Page'
+import { Link, Redirect, useParams } from 'react-router-dom'
+import { ChevronRightIcon, Flex, Text } from '@pancakeswap-libs/uikit'
+import PageLoader from 'components/PageLoader'
+import { Team as TeamType } from 'state/types'
+import useI18n from 'hooks/useI18n'
+import { getProfileContract } from 'utils/contractHelpers'
+import { TeamResponse, transformTeamResponse } from 'state/profile/helpers'
+import TeamCard from './components/TeamCard'
+import TeamHeader from './components/TeamHeader'
+
+const Team = () => {
+  const { id }: { id: string } = useParams()
+  const [is404, setIs404] = useState(false)
+  const [team, setTeam] = useState<TeamType>(null)
+  const TranslateString = useI18n()
+
+  useEffect(() => {
+    const fetchTeam = async () => {
+      try {
+        const pancakeProfileContract = getProfileContract()
+        const response = await pancakeProfileContract.methods.getTeamProfile(id).call()
+        const teamData = transformTeamResponse(response as TeamResponse)
+        setTeam(teamData)
+      } catch (error) {
+        setIs404(true)
```

Comment:
Not the best solution...

Reply:
I can't believe react router doesn't allow for route param validation using a regex ....